### ISSUE TITLE
Soft Constraints support in CVC5

### ISSUE THREAD

**Issue by singh-harish** [cvc5/cvc5#10007]
Hi developers,

Do we have support for Soft Constraints as available in Z3's Optimize class.

Z3 has AssertSoft() method in Optimize class for the same.
Optimize will try to satisfy maximum number of Soft Constraint's added to it and will ignore the ones which make's the model UNSAT.

I'm in need of one such feature in CVC5 too.
If its under development, when can I expect its release.

**Comment by martin-cs** [comment]
I'm not aware of work on soft constraints but there is some very exciting stuff on optimisation modulo theories which is going on.  I've pinged one of the people involved and hopefully they will get back to you.

**Comment by ntsis** [comment]
We are working on it. We will post here when it's available.

**Comment by singh-harish** [comment]
Hi @ntsis , any ETA or estimated release version which will include this feature.

**Comment by barrettcw** [comment]
Harish,

We are in a kind of prototyping stage for this.  My guess is the earliest
would be 6 months.

-Clark

**Comment by AntonioNapoliKTH** [comment]
Hi all, there are any news regarding this feature?

**Comment by barrettcw** [comment]
Yes, but slowly.  We had a paper in IJCAR on "generalized optimization modulo theories" which spells our our approach, including for MaxSMT.  We have recently converged on an improved syntax and have a prototype implementation.  But having this finalized and in cvc5 main is still some way off.
